azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+    this.jobLogNumFiles = this.azKabanProps.getInt("job.log.backup.index", 4);
+    this.validateProxyUser = this.azKabanProps.getBoolean("proxy.user.lock.down", false);
+    this.jobTypeManager =
+        new JobTypeManager(
+            this.azKabanProps.getString(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+                PluginManager.JOBTYPE_DEFAULTDIR),
+            this.globalProps, getClass().getClassLoader());
+    // Fetch keyStore props and use it get the KeyStore, put it in JobTypeManager
+    Props keyStoreLoadProps = this.jobTypeManager.getKeyStoreLoadProps();
+    if (keyStoreLoadProps != null) {
+      // Load HadoopSecurityManager
+      HadoopSecurityManager hadoopSecurityManager = null;
+      try {
+        final String hadoopSecurityClassName =
+                keyStoreLoadProps.getString(HadoopJobUtils.HADOOP_SECURITY_MANAGER_CLASS_PARAM);
+        final Class<?> hadoopSecurityManagerClass =
+                HadoopProxy.class.getClassLoader().loadClass(hadoopSecurityClassName);
+
+        logger.info("Loading hadoop security manager " + hadoopSecurityManagerClass.getName());
+        hadoopSecurityManager = (HadoopSecurityManager)
+                Utils.callConstructor(hadoopSecurityManagerClass, keyStoreLoadProps);
+      } catch (final Exception e) {
+        logger.error("Could not instantiate Hadoop Security Manager ", e);
+        throw new RuntimeException("Failed to get hadoop security manager!"
+                + e.getCause(), e);
+      }
+
+      final KeyStore keyStore = hadoopSecurityManager.getKeyStore(keyStoreLoadProps);
+      // Delete the cert file from disk
+      final File certFile = new File(Constants.ConfigurationKeys.CSR_KEYSTORE_LOCATION);
+      if (certFile.delete()) {
+        logger.info("Successfully deleted the cert file");
+      } else {
+        logger.error("Failed to delete the cert file");
+        throw new IOException("Failed to delete the cert file");
+      }
+      if (keyStore == null) {
+        logger.error("Failed to Prefetch KeyStore");
+        throw new IOException("Failed to Prefetch KeyStore");
+      }
+    }
+  }
+
+  public static void main(final String[] args) throws IOException, ExecutorManagerException {
+    // Redirect all std out and err messages into log4j
+    StdOutErrRedirect.redirectOutAndErrToLog();
+    // Get all the arguments
+    final String execIdStr = args[0];
+    final String projectZipName = args[1];
+    // Process Execution ID.
+    int execId = 0;
+    try {
+      execId = Integer.parseInt(execIdStr);
+    } catch (NumberFormatException ne) {
+      System.out.printf("Execution ID %s is invalid %n", args[0]);
+    }
+    System.out.printf("Execution ID = %d%n", execId);
+
+    // Setup work directories
+    final Path currentWorkingDir = Paths.get("").toAbsolutePath();
+    final Path projectDirPath = Paths.get(currentWorkingDir.toString(), PROJECT_DIR);
+    final Path jobtypePluginPath = Paths.get(currentWorkingDir.toString(), JOBTYPE_DIR);
```

Comment:
Can you please add methods like prepareProjectForLaunch and move project zip related code over there? Same for any other thing.

Reply:
You have not worked on this comment in next PR. Please fix it.

Reply:
will do.